lighthouse-core/test/gather/driver-test.js

```diff
@@ -873,73 +874,21 @@ describe('.goOnline', () => {
 describe('Multi-target management', () => {
   it('enables the Network domain for iframes', async () => {
     connectionStub.sendCommand = createMockSendCommandFn()
-      .mockResponse('Target.sendMessageToTarget', {})
-      .mockResponse('Target.sendMessageToTarget', {})
-      .mockResponse('Target.sendMessageToTarget', {});
+      .mockResponseToSession('Network.enable', 123, {})
+      .mockResponseToSession('Target.setAutoAttach', 123, {})
+      .mockResponseToSession('Runtime.runIfWaitingForDebugger', 123, {});
 
     driver._eventEmitter.emit('Target.attachedToTarget', {
       sessionId: 123,
       targetInfo: {type: 'iframe'},
     });
     await flushAllTimersAndMicrotasks();
 
-    const sendMessageArgs = connectionStub.sendCommand
-      .findInvocation('Target.sendMessageToTarget');
-    expect(sendMessageArgs).toEqual({
-      message: '{"id":1,"method":"Network.enable"}',
-      sessionId: 123,
-    });
-  });
-
-  it('enables the Network domain for iframes of iframes of iframes', async () => {
```

Comment:
The path for attaching to an iframe target is now the exact same as attaching to an iframe's iframe target, so I don't see functionality in having a test for this. I removed the test because it needed to be rewritten to continue working, and how to do so was a head-scratcher.

Reply:
> The path for attaching to an iframe target is now the exact same as attaching to an iframe's iframe target, so I don't see functionality in having a test for this.

one level deep appears to be in `gotoURL` while 2+ levels deep appears to happen in `_handleTargetAttached`, I believe?

@patrickhulce could maybe help make the call/give an idea for testing

Reply:
goToURL calls `Target.setAutoAttach` on the main target, but the first and subsequent targets will be via `_handleTargetAttached`. I suppose it was this way before too, but what's different now is that we don't have to worry about the complicated message routing - I believe this test existed just to verify that mechanism.

Reply:
yeah, it's having a test of the inductive step that would feel good, even if it's a trivial difference from the initial attachment

Reply:
though I guess what you may be saying is that since we don't keep any hierarchy of the child targets (e.g. looking at `openerId`), having a child target of a child target is indistinguishable from having two sibling child targets in terms of protocol traffic we'd care about.

That's a good point and that means the second one is a *really* trivial difference from the first, but it seems ok to include a two-child-target case, which would also provide a nice tripwire/starting point if and when we ever start treating fourth-party, etc content differently and need to track the target hierarchy

Reply:
I mostly lean on @connorjclark's side of this that there is no difference to driver in this case now and we're basically relying on chrome to do the right thing and send us nested targets, mocking Chrome doing the right thing doesn't feel super valuable since it ends up being indistinguishable to us

I feel comfortable enough with the grandchild OOPIF smoke test we have that exercises this, but a two-child-target test case seems reasonable as *the* test for this anyhow, so might as well :)